src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -398,7 +398,13 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
 
                     // SPEC: When this is used in a primary-expression within an instance method or instance accessor
                     // SPEC: of a struct, it is classified as a variable. 
-                    if (!thisref.Type.IsValueType || RequiresReturnableReference(valueKind))
+
+                    // "this" is only readable if we are not in a struct. RValueOnly is checked at the beginning of this method
```

Comment:
Should fix this comment. Now that I read it seems ambiguous. The comment should say: If we are not in a struct "this" is only readable. Since we are hear, we need more than readable - RValueOnly is checked at the beginning of this method.